### ISSUE TITLE
Request: Improve error highlighting

### ISSUE THREAD

**Issue by sloria** [AlessandroYorba/Sierra#4]
When using sierra in the terminal, the highlighting color for errors has very low contrast against the text, making the text very hard to read.

![tmux](https://cloud.githubusercontent.com/assets/2379650/18072572/6ffed588-6e2d-11e6-9415-659ee90e322c.png)

Perhaps you could use colors that are similar to search result highlighting, like "BaseModel" in this screenshot:

![tmux](https://cloud.githubusercontent.com/assets/2379650/18072597/8e930e9c-6e2d-11e6-9d5f-3a60c1a3b4ad.png)

**Comment by AlessandroYorba** [comment]
You're right.  Both the foreground and background colors you have there are not the intended Error colors.
The foreground color for Errors should be the same dark gray or black depending on the sierra mode being used. And the red should always be cterm color `131` and hex color `#af5f5f`.

Can you please, open vim with sierra running and type:
`:hi Error`
and read my back the output

As far as I can tell Error are displaying correctly on my end. 
I'm using vim 7.5.1831 with sierra v3.0.0 inside of iTerm build 3.0.7 Here are some examples.  
![1](https://cloud.githubusercontent.com/assets/11221489/18078862/989f7ece-6e42-11e6-98e9-e77aba5f65b5.png)
![2](https://cloud.githubusercontent.com/assets/11221489/18078866/9f0d687a-6e42-11e6-9f40-4a5eeeff5e4b.png)
![3](https://cloud.githubusercontent.com/assets/11221489/18078867/a158b3d2-6e42-11e6-8906-6a9e1cc86e71.png)
![4](https://cloud.githubusercontent.com/assets/11221489/18078869/a3cab494-6e42-11e6-95f7-38838ae3f3ad.png)
![5](https://cloud.githubusercontent.com/assets/11221489/18078874/a5acf5ce-6e42-11e6-969f-6cd83340b12b.png)

**Comment by sloria** [comment]
Here is the output of `:hi Error`:

![tmux](https://cloud.githubusercontent.com/assets/2379650/18088632/a78211ee-6e89-11e6-96cd-6e0597e0586b.png)

I am using vim 7.4.2275 in iterm 3.0.7 with sierra installed via [vim-colorschemes](https://github.com/flazz/vim-colorschemes). I also tried installing the latest master version of sierra from this repo, but I still see the same problem.

**Comment by sloria** [comment]
I even tried changing my terminal colors to use sierra_Midnight.Xresources, and that actually made it worse!

![tmux](https://cloud.githubusercontent.com/assets/2379650/18088898/2d3a70c8-6e8b-11e6-89c9-d4c6660474b3.png)

![tmux](https://cloud.githubusercontent.com/assets/2379650/18088910/3baf0e98-6e8b-11e6-98bf-70d32963a6de.png)

**Comment by AlessandroYorba** [comment]
@sloria that was a good idea using the Xresources. Even though it is obviously broken when using the midnight palette, it leads me to believe that something in Vim is referencing the terminal colors and not sierra's internal highlight Error colors. My guess is that a plugin is responsible, but that's just my guess.  

Try changing the Xresources color9 to the value below or from the iTerm preferences > profiles > colors > change Bright Red #9 to hex #000000

! red
*.color9:       #000000

**Comment by sloria** [comment]
@AlessandroYorba Ah, your hypothesis is confirmed: when I changed Bright Red #9 to #000000, the Error highlight color becomes black.

I am unsure what the culprit is; I have only seen this happen when using sierra.

**Comment by sloria** [comment]
FWIW, I'm seeing the same issue when I use Alduin.

**Comment by AlessandroYorba** [comment]
@sloria after the change, was the foreground color black or the background black? Same thing with Alduin and Sierra probably because color1 and color 9 share the same value. Whatever it is that is calling the Errors it's using both color1 and color9 for foreground and background. What was the terminal theme used in your first screenshot?

**Comment by sloria** [comment]
The background color became black.

Here is the theme I am using in the screen shot: https://gist.github.com/sloria/655377de989353d8d9083a01eacaf85b

**Comment by AlessandroYorba** [comment]
Thank you @sloria , so far I believe that a plugin is responsible.  

ANSI Colors
`.color1` used for foreground 
`.color9` used for the background

Since I cannot override the plugin from my end, I can offer a workaround. Here are the colors for your iTerm theme that should show errors similar to the Visual highlight you liked in your first post.  

`*.color1: #dfdfaf`
`*.color9: #875f5f`

**Comment by sloria** [comment]
@AlessandroYorba I tried disabling all plugins except for `vim-colorschemes` and the syntax checker ([khuno](https://github.com/alfredodeza/khuno.vim)) and commented all the settings in my vimrc, and the issue still persists. Even if I change the syntax-checking plugin to Syntastic, the colors are still incorrect, so it seems unlikely that a plugin is the culprit. Though I'm open to ideas for further diagnosing the issue.

**Comment by AlessandroYorba** [comment]
@sloria good call. I will continue to look at this via documentation and I'll also check other more popular and older color schemes to see if this issue has come up. 
Let's leave this ticket open for others who might have a solution.

**Comment by sloria** [comment]
In addition to iTerm2, I have confirmed the issue also exists in Terminal.app. The issue also exists regardless of whether I am using zsh or bash.

**Comment by AlessandroYorba** [comment]
Let's see if we can find out what highlighting group vim is using. Add this to your .vimrc:

``` VimL
nmap <C-S-P> :call <SID>SynStack()<CR>
function! <SID>SynStack()
  if !exists("*synstack")
    return
  endif
  echo map(synstack(line('.'), col('.')), 'synIDattr(v:val, "name")')
endfunc
```

Then pull up the file and once the Error highlighting is active, place your cursor over any word within the Error. Then from your keyboard click:

 Control + Shift + p

At the bottom of vim in the messaging area all of the highlighting groups should be listed. Let's see what highlighting groups were found.

**Comment by sloria** [comment]
The output after hitting `C-S-P` is

```
['pythonString']
```
